### DISCUSSION TITLE
Change BASE_URL

### DISCUSSION THREAD

**Discussion by mnaranjorion** [airflow-helm/charts#657] [Questions & Answers] [answered]
Hello,

I've been trying to set the 'base_url' to be http://localhost:8080/airflow instead of http://localhost:8080/.

To make the change I have included in the configuration file extra call in the '_**helm**_' statement with _--values_:

```
extraEnv: |
   - name: AIRFLOW__WEBSERVER__BASE_URL
     value: 'http://localhost:8080/airflow'
```

However, when I do this, when I pull up the Pod from the webserver, it does not perform the liveness and readiness action well, as it always goes against **/health** instead of **/airflow/health**.

To correct this, I have tried modifying the paths of livenessProb and readinessProb. If I modify them at the root level of the configuration file, nothing happens, and inside the webserver service I can't because '_httpGet_' is not allowed parameter.

Has anyone made the change of base_url? How to do it so that the Pod is lifted and makes correctly the passage of ReadinessProb and LivenessProb?

Thank you very much!

**Comment by thesuperzapper** [comment, marked as answer]
@mnaranjorion is there a specific reason you need to set `AIRFLOW__WEBSERVER__BASE_URL` using `airflow.extraEnv` rather than `airflow.config`?

You won't have this problem if you use `airflow.config` to set `AIRFLOW__WEBSERVER__BASE_URL`.

----

To make the webserver probes work when setting `AIRFLOW__WEBSERVER__BASE_URL` with `airflow.extraEnv`, we will have to change [the current `httpGet` probe](https://github.com/airflow-helm/charts/blob/main/charts/airflow/templates/webserver/webserver-deployment.yaml#L119-L123) to an `exec` liveness probe. This is because  `airflow.extraEnv` can set env-vars from Secrets/ConfigMaps, so we must actually call the`$AIRFLOW__WEBSERVER__BASE_URL` env-var directly.

(This solution is similar to [the current flower probes](https://github.com/airflow-helm/charts/blob/420eae29c454f6e7e6a7837706ca2e6c0fe792b8/charts/airflow/templates/flower/flower-deployment.yaml#L129), however, I actually see a bug to fix in that code also, because we only check if `AIRFLOW__CELERY__FLOWER_BASIC_AUTH` is set using `flower.basicAuthSecret` or `airflow.config` and forgot to check `airflow.extraEnv`).

---

Give all this, it may just be easier to prevent people from setting `AIRFLOW__WEBSERVER__BASE_URL` using a [`values-validaiton.tpl`](https://github.com/airflow-helm/charts/blob/420eae29c454f6e7e6a7837706ca2e6c0fe792b8/charts/airflow/templates/_helpers/validate-values.tpl#L70-L72). (Probably `AIRFLOW__CELERY__FLOWER_BASIC_AUTH` also)

**Comment by mnaranjorion** [comment]
Hello,

Well, because many times you have the answer in front of you and you go around it a thousand times without seeing it. 

Indeed, this type of configuration (change the value in `airflow.config`) is the correct one and now **Liveness Probe** and **Readiness Probe** go against `/airflow`.

Thank you very much for the clarification @thesuperzapper.

Best regards.
